arch/risc-v/src/mpfs/mpfs_entrypoints.c

```diff
@@ -88,6 +100,21 @@ void mpfs_jump_to_app(void)
   __asm__ __volatile__
     (
       "csrr a0, mhartid\n"                   /* Hart ID */
+#ifdef CONFIG_MPFS_BOARD_PMP
+      "li   t1, %0\n"                        /* Size of hart's stack */
+      "mul  t0, a0, t1\n"                    /* Hart stack base */
+      "add  t0, t0, t1\n"                    /* Hart stack top */
+      "la   sp, g_mpfs_boot_stacks\n"        /* Stack area base */
+      "add  sp, sp, t0\n"                    /* Set stack pointer */
+      "call mpfs_board_pmp_setup\n"          /* Run PMP configuration */
+      "csrr a0, mhartid\n"                   /* Restore hartid */
+#else
+      "li   t0, -1\n"                        /* Open the whole SoC */
+      "csrw pmpaddr0, t0\n"
+      "li   t0, %0\n"                        /* Grant RWX permissions */
+      "csrw pmpcfg0, t0\n"
+      "csrw pmpcfg2, zero\n"
+#endif
```

Comment:
@jlaitine This opart maybe doable via ToC configuration (like g_hart_use_sbi), do you think that would be a more scalable solution ?

This way SMP payloads could disable PMP altogether.